### PR TITLE
Prepare 7.2.4 release

### LINKED ISSUE #832: LoadError: cannot load such file -- i18n/core_ext/hash  in lib/chewy.rb:17
As of 1.9.1, the i18n gem no longer contains `i18n/core_ext/hash` but this is being required in `lib/chewy.rb:17` which of course throws an error whenever trying to do something in Rails.

This stacktrace is from our CI, just to show the error being thrown.

```sh
#!/bin/bash -eo pipefail
bundle exec rails db:setup
rails aborted!
LoadError: cannot load such file -- i18n/core_ext/hash
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `require'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `block in require'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:299:in `load_dependency'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `require'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/chewy-7.2.3/lib/chewy.rb:17:in `<top (required)>'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `require'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `block in require'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:299:in `load_dependency'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `require'
/home/circleci/my_engine/lib/my_engine/engine.rb:16:in `<top (required)>'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `require'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `block in require'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:299:in `load_dependency'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `require'
/home/circleci/my_engine/lib/my_engine.rb:3:in `<top (required)>'
/home/circleci/my_engine/test/dummy/config/application.rb:7:in `<top (required)>'
/home/circleci/my_engine/test/dummy/Rakefile:6:in `require_relative'
/home/circleci/my_engine/test/dummy/Rakefile:6:in `<top (required)>'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:326:in `load'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:326:in `block in load'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:299:in `load_dependency'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:326:in `load'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.4.4/lib/rails/tasks/engine.rake:5:in `block (2 levels) in <top (required)>'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.4.4/lib/rails/tasks/engine.rake:4:in `block in <top (required)>'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.4.4/lib/rails/tasks/engine.rake:85:in `<top (required)>'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:326:in `load'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:326:in `block in load'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:299:in `load_dependency'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:326:in `load'
/home/circleci/my_engine/Rakefile:5:in `<top (required)>'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:326:in `load'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:326:in `block in load'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:299:in `load_dependency'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:326:in `load'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.4.4/lib/rails/commands/rake/rake_command.rb:20:in `block in perform'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.4.4/lib/rails/commands/rake/rake_command.rb:18:in `perform'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.4.4/lib/rails/command.rb:50:in `invoke'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.4.4/lib/rails/commands.rb:18:in `<top (required)>'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `require'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `block in require'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:299:in `load_dependency'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `require'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/railties-6.1.4.4/lib/rails/engine/commands.rb:9:in `<top (required)>'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `require'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `block in require'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:299:in `load_dependency'
/home/circleci/my_engine/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.4.4/lib/active_support/dependencies.rb:332:in `require'
bin/rails:16:in `<main>'
Tasks: TOP => load_app
(See full trace by running task with --trace)

Exited with code exit status 1
CircleCI received exit code 1
```

We've locked i18n to 1.8.11 until this is resolved. Also might be worth looking at https://github.com/ruby-i18n/i18n/issues/603 as 1.9.0 of the i18n gem was released with the same issue.

## Steps to reproduce the problem

* Upgrade i18n to 1.9.1
* Try to run `rails`, `rubocop` etc and see the error thrown

## Version Information

* Chewy version: 7.2.3
* Elasticsearch version: 7.13.0
* Ruby version: 3.0.3
* Rails version: 6.1.4.4

### PR DESCRIPTION
**Pull request by rabotyaga**
- [Remove `deep_dup` backport](https://github.com/toptal/chewy/pull/834/commits/bb9094727cb5865bddae98a4ca6c1c7361e21e3d)
- [Replace `i18n/core_ext/hash` with `active_support/core_ext/hash/keys` (`deep_symbolize_keys`)](https://github.com/toptal/chewy/pull/834/commits/1c09384a10455e6c7e9c8c76c2a2c6da17d572a6)
  - Fixes #832 
  - See https://github.com/toptal/chewy/pull/833
- [Add Ruby 3.1 & ActiveRecord 7.1 to CI matrix](https://github.com/toptal/chewy/pull/834/commits/f4dbd9332e870889a1504234094620e4359e1415)
  - See https://github.com/toptal/chewy/pull/831
- [Bump version to 7.2.4](https://github.com/toptal/chewy/pull/834/commits/eb785453beadc1c3d686f0948ccda740fee92672)

### CODE REVIEW

File: spec/support/active_record.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 require 'database_cleaner'
 
-ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: 'file::memory:?cache=shared', pool: 10)
+ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:', pool: 10)
```
Comment (rabotyaga) [inline spec/support/active_record.rb:3]: See https://github.com/rails/rails/blob/6e721d79ba9458317c93bb02590bbaa25c5a3649/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L29

**Review by mrzasa — APPROVED** [review]

**Comment by gr66** [comment]
Please fix the readme file like in pr [#831](https://github.com/toptal/chewy/pull/831)

**Comment by rabotyaga** [comment]
Hey @gr66 👋 !
Sorry, didn't understand your message. Could you please elaborate on it a little?

**Comment by gr66** [comment]
> Hey @gr66 👋 ! Sorry, didn't understand your message. Could you please elaborate on it a little?

it [#831](https://github.com/toptal/chewy/pull/831) @petergoldstein wrote > This PR adds Ruby 3.1 and ActiveRecord 7.0
but in readme file still > 5.2, 6.0, 6.1 Active Record versions are supported by all Chewy versions. 
and > Add Ruby 3.1 & ActiveRecord 7.1 to CI matrix
I think a lot of people who use your gem are waiting for support for these versions, and not everyone keeps track of what's going on in the PR. And they are guided by the description of changes or the description in the readme of the new version.

**Comment by rabotyaga** [comment]
Hey @gr66 👋 !
I see now, thanks for your suggestion.
We've added AR 7.1 & Ruby 3.1 to CI matrix, that's true, however, we didn't test chewy extensively in production environments with those versions. I think it's too early to officially declare we're fully supporting them. But enthusiasts can try using them.

**Pull request merged**